src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -123,13 +120,121 @@ public static ControlFlowGraph Create(IOperation body, ControlFlowGraph.Region e
             builder._regionMap.Free();
             builder._labeledBlocks?.Free();
 
-            return new ControlFlowGraph(blocks.ToImmutableAndFree(), region, methods.ToImmutableAndFree(), methodsMap.ToImmutable());
+            return new ControlFlowGraph(body, ToImmutableBlocks(blocks), region, methods.ToImmutableAndFree(), methodsMap.ToImmutable());
+        }
+
+        private static ImmutableArray<FlowAnalysis.BasicBlock> ToImmutableBlocks(ArrayBuilder<BasicBlock> blocksBuilder)
+        {
+            var builder = ArrayBuilder<FlowAnalysis.BasicBlock>.GetInstance(blocksBuilder.Count);
+            var branchMap = PooledDictionary<(BasicBlock source, BasicBlock destination, bool isFallThrough), ControlFlowBranch>.GetInstance();
+            var ordinalToBlockMap = PooledDictionary<int, FlowAnalysis.BasicBlock>.GetInstance();
+
+            // Pass 1: Iterate through blocksBuilder to create basic blocks.
+            foreach (BasicBlock blockBuilder in blocksBuilder)
+            {
+                var block = new FlowAnalysis.BasicBlock(blockBuilder.Kind,
+                    blockBuilder.GetImmutableStatements(),
+                    blockBuilder.InternalConditional.Condition,
+                    blockBuilder.InternalNext.Value,
+                    getConditionKind(blockBuilder),
+                    blockBuilder.Ordinal,
+                    blockBuilder.IsReachable, blockBuilder.Region);
+                builder.Add(block);
+                ordinalToBlockMap.Add(block.Ordinal, block);
+            }
+
+            // Pass 2: Create control flow branches with source and destination info and
+            //         update the branch information for the created basic blocks.
+            for (int i = 0; i < blocksBuilder.Count; i++)
+            {
+                BasicBlock blockBuilder = blocksBuilder[i];
+                FlowAnalysis.BasicBlock block = builder[i];
+
+                ControlFlowBranch successor = getFallThroughSuccessor(blockBuilder);
+                ControlFlowBranch conditionalSuccessor = getConditionalSuccessor(blockBuilder);
```

Comment:
>=> [](start = 79, length = 2)

I do not believe expression  bodies are actually provide any advantage when the expression is non trivial and long. No need to change.